pkg/cmd/flags/policy.go

```diff
@@ -0,0 +1,219 @@
+		if err != nil {
+			return nil, err
+		}
+
+		filterFlags := make([]*filterFlag, 0)
+
+		if p.Scope == nil {
+			return nil, errfmt.Errorf("scope cannot be empty")
+		}
+		// scope
+		for _, s := range p.Scope {
+			if s == "global" {
+				continue
+			}
+
+			operatorIdx := strings.IndexAny(s, "=!<>")
+
+			err := validateScope(s[:operatorIdx])
+			if err != nil {
+				return nil, err
+			}
+
+			// todo, validar um operdor não existente
+			// if there is no operator, an error must be returned?
+			// there is follow, container, c though
+
+			filterFlags = append(filterFlags, &filterFlag{
+				full:              fmt.Sprintf("%s", s),
+				filterName:        fmt.Sprintf("%s", s[:operatorIdx]),
+				operatorAndValues: fmt.Sprintf("%s", s[operatorIdx:]),
+				scopeIdx:          i,
+			})
+		}
+
+		for _, r := range p.Rules {
+			err := validateEvent(p.Name, r.Event)
+			if err != nil {
+				return nil, err
+			}
+
+			filterFlags = append(filterFlags, &filterFlag{
+				full:              fmt.Sprintf("event=%s", r.Event),
+				filterName:        "event",
+				operatorAndValues: fmt.Sprintf("=%s", r.Event),
+				scopeIdx:          i,
+			})
+
+			for _, f := range r.Filter {
+				operatorIdx := strings.IndexAny(f, "=!<>")
+				// if there is no operator, an error must be returned?
+				// there is follow, container, c though
+
+				// args
+				if strings.HasPrefix(f, "args.") {
+					// do I validate the argument?
+					filterFlags = append(filterFlags, &filterFlag{
+						full:              fmt.Sprintf("%s.%s", r.Event, f),
+						filterName:        fmt.Sprintf("%s.%s", r.Event, f[:operatorIdx]),
+						operatorAndValues: fmt.Sprintf("%s", f[operatorIdx:]),
+						scopeIdx:          i,
+					})
+
+					continue
+				}
+
+				// retval
+				if strings.HasPrefix(f, "retval.") {
+					filterFlags = append(filterFlags, &filterFlag{
+						full:              fmt.Sprintf("%s.retval.%s", r.Event, f),
+						filterName:        fmt.Sprintf("%s.retval.%s", r.Event, f[:operatorIdx]),
+						operatorAndValues: fmt.Sprintf("%s", f[operatorIdx:]),
+						scopeIdx:          i,
+					})
+					continue
+				}
+
+				err = validateContext(f[:operatorIdx])
+				if err != nil {
+					return nil, err
+				}
+
+				// context
+				filterFlags = append(filterFlags, &filterFlag{
+					full:              fmt.Sprintf("%s.context.%s", r.Event, f),
+					filterName:        fmt.Sprintf("%s.context.%s", r.Event, f[:operatorIdx]),
+					operatorAndValues: fmt.Sprintf("%s", f[operatorIdx:]),
+					scopeIdx:          i,
+				})
+			}
+		}
+
+		parsedMap[i] = filterFlags
+	}
+
+	// empty?
+
+	return createFilterScopes(parsedMap)
+}
+
+func validatePolicy(p policy.Policy) error {
+	if p.Name == "" {
+		return errfmt.Errorf("policy name cannot be empty")
+	}
+
+	if p.Description == "" {
+		return errfmt.Errorf("policy %s: description cannot be empty", p.Name)
+	}
+
+	if p.Scope == nil || len(p.Scope) == 0 {
+		return errfmt.Errorf("policy %s: scope cannot be empty", p.Name)
+	}
```

Comment:
We must also err when `p.Scope == "global"` and `len(p.Scope) > 1`. And of course when we come across other invalid constructions.